src/main/scala/org/dbpedia/spotlight/wikistats/JsonPediaParser.scala

```diff
@@ -76,6 +80,41 @@ class JsonPediaParser(inputWikiDump:String, lang:String)
     rawXmls.map(p => p._2.toString)
   }
 
+  /*
+    Get Redirects from the wiki dump
+    Input:   - Base dataframe consiting of wiki data
+    Output:  - RDD with Redirect source and target
+   */
+  def redirectsWikiArticles(): RDD[(String,String)] = {
+
+    dfWikiRDD.select("wikiTitle","type","redirect")
+    .rdd
+    .filter(row => row.getString(1)== "REDIRECT")
+    .map(row => (row.getString(0),row.getString(2)))
+
+  }
+
+  /*
+    Method to resolve transitive dependencies for the redirects.
+    Input:  - None
+    Output: - RDD with the resolved wiki uri and surface form
+   */
+  def getResolveRedirects(): RDD[(String,String)] = {
+
+    val rddRedirects = redirectsWikiArticles()
+    var linkMap = sc.accumulableCollection(HashMap[String,String]())
```

Comment:
Ive not sued `accumulableCollection ` before.
Is there an advantage in this case over just generating an rdd with the tuples, then collecting it and transforming it into a hashmap?